lib/pbench/server/database/models/users.py

```diff
@@ -119,7 +119,10 @@ def delete(username):
         """
         try:
             user_query = Database.db_session.query(User).filter_by(username=username)
-            user_query.delete()
+            rows_deleted = user_query.delete()
+            if rows_deleted == 0:
+                return False
```

Comment:
Rather than returning `False` here, I would recommend raising an appropriate exception (perhaps `ValueError` or a custom exception derived from it) -- if you give it a suitable message text, then the `except` block in `pbench-user-delete` will catch it print the message.

(And, of course, I would recommend doing the test before the `delete()`, based on what the filtered query returned.)

(And, I would add a sanity check there that the filtered query returned exactly one row....)